app/src/main/kotlin/org/akanework/gramophone/logic/GramophoneExtensions.kt

```diff
@@ -98,6 +99,25 @@ fun MediaItem.getFile(): File? {
     return getUri()?.toFile()
 }
 
+fun MediaItem.getBitrate(context: Context): Long {
+    val retriever = MediaMetadataRetriever()
+    return try {
+        val filePath = getFile()?.path
+        if (filePath != null) {
+            retriever.setDataSource(filePath)
+        } else {
+            val uri = requestMetadata.mediaUri ?: return 0
+            retriever.setDataSource(context, uri)
+        }
```

Comment:
Is this else path really needed?

Reply:
Yes, the `else` path is needed to handle cases where the media does not have a local file path but instead has a valid `content://` URI, which is common for media retrieved from Android's MediaStore. This ensures that `MediaMetadataRetriever `can still extract metadata when the file path is unavailable.

Reply:
In general, yes, but in Gramophone's case _every_ media item should have a file path because we query the real path from MediaStore when querying songs. Many things in the app assume this, and it's fine to do so here as well. Unless it doesn't work without the else path, which would mean I missed something, which can happen. Hence: is this else _really_ needed? Or does it work without? 😉

Reply:
Let me check it out

Reply:
Great! Looks like it still works fine without the else path. I'll keep it simplified unless we run into any unexpected edge cases. Thanks for the review! 🚀